app/models/manageiq/providers/oracle/infra_manager.rb

```diff
@@ -0,0 +1,98 @@
+class ManageIQ::Providers::Oracle::InfraManager < ManageIQ::Providers::InfraManager
+  require_nested :RefreshWorker
+  require_nested :RefreshParser
+  require_nested :Host
+  require_nested :Refresher
+  require_nested :Template
+  require_nested :Vm
+
+  def self.ems_type
+    @ems_type ||= "oraclevm".freeze
+  end
+
+  def self.description
+    @description ||= "Oracle Virtualization Manager".freeze
+  end
+
+  def self.default_blacklisted_event_names
+    []
+  end
+
+  def supports_port?
+    true
+  end
+
+  def supported_auth_types
+    %w(default)
+  end
+
+  def supports_authentication?(authtype)
+    supported_auth_types.include?(authtype.to_s)
+  end
+
+  def self.raw_connect(server, port, username, password)
+    require 'fog/oracle'
+
+    url = "https://#{server}:#{port}"
```

Comment:
I think something like this:

```ruby
URI::HTTP.build(:host => server, :port => port)
```

For example:
```ruby
irb(main):004:0> server = "::1"
=> "::1"
irb(main):005:0> port = 1234
=> 1234
irb(main):007:0> uri = URI::HTTP.build(:host => server, :port => port)
=> #<URI::HTTP http://[::1]:1234>
irb(main):008:0> uri.hostname
=> "::1"
irb(main):009:0> uri.host
=> "[::1]"
irb(main):010:0> uri.to_s
=> "http://[::1]:1234"
```

It's better to pass along the URI to fog instead of giving them a string they have to parse and do the wrong thing with.

URI#hostname is used for socket connections (strips the [])
URI#host is used for nearly everything else (host http request heade)